src/commonMain/kotlin/com/airthings/lib/logging/facility/FileLoggerFacility.kt

```diff
@@ -128,19 +129,15 @@ class FileLoggerFacility(
         notifier = null,
     )
 
-    private val io: PlatformFileInputOutput = PlatformFileInputOutputImpl()
+    private val io: PlatformFileInputOutput = DelegateFileInputOutput(
+        folder = baseFolder,
+        io = PlatformFileInputOutputImpl(),
+        onFolderMissing = {
+            notifier?.onLogFolderInvalid(it) ?: DelegateFileInputOutput.reportMissingFolder(it)
```

Comment:
@jstarczewski I kept the previous logic where an exception is thrown, for backward-compatibility.